src/system/SystemLayer.cpp

```diff
@@ -31,5 +31,16 @@ CHIP_ERROR Layer::ScheduleLambdaBridge(LambdaBridge && bridge)
     return lReturn;
 }
 
+#if CHIP_SYSTEM_CONFIG_USE_LWIP || CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT
+CHIP_ERROR LayerFreeRTOS::RunOnMatterContext(std::function<CHIP_ERROR()> func)
```

Comment:
Maybe `RunWithMatterContextLock` or similar, since `RunOn` seems to imply thread loop transfer and this is not doing that.

Some description on why this is safe/does not deadlock would be useful.

Reply:
Changed the name of the function and added some comments on the function of using non-blocking function as the input to avoid dead lock.